### PR TITLE
Improve display of skipped tasks

### PR DESCRIPTION
**Pull request by AlanGreene**
# Changes

> [!NOTE]
> This PR addresses a current gap in functionality, surfacing important information to the user to help understand the behaviour of a given pipeline / pipeline run.
>
> It is a work in progress of one potential tactical change to improve the display of skipped tasks in the short term.
> The design and/or approach may change drastically before it is merged.
>
> However, a much larger and far-reaching redesign of the PipelineRun details page is still planned, see https://github.com/tektoncd/dashboard/issues/2306 for details of some of the identified aspects. That change will likely involve updates to the navigation structure of the task list, the display of logs, and may also impact the way the 'skipped' status is surfaced.

Resolves https://github.com/tektoncd/dashboard/issues/2293
Resolves https://github.com/tektoncd/dashboard/issues/1445

When a task is skipped ensure this is communicated to the user instead of leaving it showing as 'pending'.

In the case of when expressions, surface details of the when expression so the user can undertsand the reason for the task being skipped. Also remove elements that will never have content in this case such as the pod tab on the TaskRun details view, and replace the log container with a message indicating that the task was skipped and no logs will be available.

Update the status icons and labels to differentiate between pending / not run and skipped.

Apply similar changes for step-level when expressions which rely on the step actions feature being enabled.

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [ ] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [ ] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (new features, significant UI changes, API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: packages/components/src/components/DetailsHeader/DetailsHeader.jsx [hunk 1]
```diff
@@ -126,11 +133,16 @@ export default function DetailsHeader({
   if (type === 'taskRun') {
     ({ reason: reasonToUse, status: statusToUse } = getStatus(taskRun));
     statusLabel =
-      reasonToUse ||
-      intl.formatMessage({
-        id: 'dashboard.taskRun.status.pending',
-        defaultMessage: 'Pending'
-      });
+      reason === 'tkn-dashboard:skipped'
```
Comment (briangleeson) [inline packages/components/src/components/DetailsHeader/DetailsHeader.jsx:136]: Would it be worth storing this string as a constant? As it's used across quite a few files

**Review by briangleeson — COMMENTED** [review]
One question but otherise lgtm

File: packages/components/src/components/DetailsHeader/DetailsHeader.jsx (hunk 1 above)
Comment (AlanGreene) [reply packages/components/src/components/DetailsHeader/DetailsHeader.jsx]: yep, done

**Review by briangleeson — APPROVED** [review]
/lgtm

**Pull request merged**